python/ray/serve/_private/logging_utils.py

```diff
@@ -225,8 +227,44 @@ def record_factory(*args, **kwargs):
     logger.addHandler(file_handler)
 
 
+def configure_component_memory_logger(
+    component_name: str,
+    component_id: str,
+    component_type: Optional[ServeComponentType] = None,
+):
+    """Configures the memory logger for this component.
+
+    Does nothing if RAY_SERVE_ENABLE_MEMORY_PROFILING is disabled.
+    """
+
+    if RAY_SERVE_ENABLE_MEMORY_PROFILING:
+        try:
+            import memray
+
+            memray_file_name = get_component_log_file_name(
+                component_name=component_name,
+                component_id=component_id,
+                component_type=component_type,
+                suffix="_memray.bin",
+            )
+
+            return memray.Tracker(memray_file_name, native_traces=True).__enter__()
```

Comment:
weird pattern, document why it's written this way

Reply:
I changed the pattern:

```python
tracker = memray.Tracker(...)
tracker.__enter__()
```

Reply:
I restored the pattern, and I added a comment explaining why its written that way.